### ISSUE TITLE
Name field with a resource type of "*" generates "AsResourceName" property

### ISSUE THREAD

**Issue by jskeet** [googleapis/gapic-generator-csharp#205]
Something like this:

```proto
message FooRequest {
  // The name of the Bar to do something with
  string name = 1 [
    (google.api.field_behavior) = REQUIRED,
    (google.api.resource_reference) = {
      type: "*"
    }
  ];
```

Generates a partial class with a property of `FooRequest.AsResourceName`. It should probably just be `FooRequest.ResourceName`.

**Comment by jskeet** [comment]
(I'm going to attempt to fix this myself. Let's see how it goes...)

**Comment by jskeet** [comment]
This was fixed in #206.
